### ISSUE TITLE
subdomain / port issue

### ISSUE THREAD

**Issue by mjhea0** [dokku-alt/dokku-alt#236]
When I set this up on Digital Ocean, it's trying to use a subdomain rather than an IP and a Port. From what I can tell, it looks like the configuration is not updating the nginx config correctly.

Has anyone come across this issue?

**Comment by ayufan** [comment]
Dokku-alt doesn't by default publish applications, instead it creates default nginx configuration with subdomains. I'll think about using: `xip.io` it will solve some potential problems with domains when setting up dokku-alt.

**Comment by mjhea0** [comment]
This is the problem. The nginx file is not being created properly. 

I followed these steps:

1. Install Dokku-Alt - `sudo bash -c "$(curl -fsSL https://raw.githubusercontent.com/dokku-alt/dokku-alt/master/bootstrap.sh)"`
1. Point your browser to http://<ip>:2000/ and finish configuration.

In the config, I did not specify a domain. I just left it alone. However, the nginx file did not get setup correctly for utilizing the IP:PORT combo.

Would you like to see screenshots?

**Comment by ayufan** [comment]
It will not, because it will use as VHOST server's hostname.

**Comment by ayufan** [comment]
If you don't have domain, you can try to:
```bash
sudo su dokku
echo $(hostname --ip-address).xip.io  > ~/VHOST
```

**Comment by mjhea0** [comment]
Ahhh. Okay. That makes sense. If you don't enter a subdomain on the config, it won't create the VHOST file. You may want to update the documentation.

**Comment by ayufan** [comment]
Yes. I'll have to switch it to use xip.io. It makes it really easy to test :)

**Comment by mjhea0** [comment]
Thanks, @ayufan. I should have a nice blog post coming out very soon on https://realpython.com/blog. Best!

**Comment by ayufan** [comment]
Great to hear :) Thanks!

On Monday, April 6, 2015, Michael Herman <notifications@github.com> wrote:

> Thanks, @ayufan <https://github.com/ayufan>. I should have a nice blog
> post coming out very soon on https://realpython.com/blog. Best!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dokku-alt/dokku-alt/issues/236#issuecomment-90053169>.
>


-- 
Kamil Trzciński

ayufan@ayufan.eu
www.ayufan.eu

**Comment by ingro** [comment]
Sorry if I post here but I was in the same situation as @mjhea0 , my VM doesn't have a domain associated yet because I'm using it with development purposes.

I tried to use xip.io as @ayufan suggested but I keep getting nginx's default page instead of my app. Any idea of what I can do to solve the problem? Thank you!

**Comment by mjhea0** [comment]
@ingro yes, there seems to be an nginx issue with how dokku-alt configures it based on the domain.

**Comment by yshahin** [comment]
@ayufan what if I have a subdomain like my_app.herokuapp.net it still doesnt work and tries to setup nginx as if the domain is app1.my_app.herokuapp.net

**Comment by dearwish** [comment]
Hi there,

I installed the dokku-alt on a fresh Ubuntu VM on Microsoft Azure.
In the domain configuration I added my custom domain name, but the issue that the virtual subdomains are not resolving to my applications.
For example when I publish a node-js-sample app as ```sample```, I am expecting that it will be accessible as http://sample.mydomain.com, but it is not accessible. 
I tried to install a new VM and use the option with http://mydomain.com:<app-port>, but it is also not working.
@ayufan Is there some resolution for this issue? Maybe some manual configuration I need to do?
